src/sentry/tasks/update_user_reports.py

```diff
@@ -19,16 +20,43 @@ def update_user_reports(**kwargs):
         group__isnull=True, environment__isnull=True, date_added__gte=now - timedelta(days=1)
     )
 
-    count = 0
-    updated = 0
-    for report in user_reports:
-        count += 1
-        event = eventstore.get_event_by_id(report.project_id, report.event_id)
-        if event:
-            report.update(group_id=event.group_id, environment=event.get_environment())
-            updated += 1
+    # We do one query per project, just to avoid the small case that two projects have the same event ID
+    project_map = {}
```

Comment:
You could use `collections.defaultdict` to save a setdefault() call, but it is more of a🏌 level comment.